File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function solution(numbers) {
    const primes = new Set();

    prime("", numbers);

    return [...primes].length;

    function prime(num, remain) {
        if (num && isPrime(Number(num))) {
            primes.add(Number(num));
        }

        for (let i = 0; i < remain.length; i++) {
            prime(
                num + remain[i],
                remain.slice(0, i).concat(remain.slice(i + 1))
            );
        }
    }

    function isPrime(num) {
        if (num === 0 || num === 1) return false;
        if (num === 2) return true;

        for (let i = 2; i <= parseInt(Math.sqrt(num)); i++) {
            if (num % i === 0) return false;
        }

        return true;
    }
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
import math
from itertools import permutations


def solution(numbers):
    primes = set()

    for l in range(1, len(numbers) + 1):
        for p in permutations(numbers, l):
            num = int(''.join(p))

            if is_prime(num):
                primes.add(num)

    return len(primes)

def is_prime(n):
    if n <= 1:
        return False

    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False

    return True